File: ios/GuardianApp/Models/Alert.swift
import Foundation

struct Alert: Codable, Identifiable {
    let id: String
    let severity: Severity
    let title: String
    let message: String
    let timestamp: Date
    let account: String
    let isRead: Bool

    enum Severity: String, Codable {
        case critical = "CRITICAL"
        case high = "HIGH"
        case medium = "MEDIUM"
        case low = "LOW"
    }
}

struct AlertListResponse: Codable {
    let alerts: [Alert]
    let total: Int
}

File: android/app/src/main/kotlin/com/aws/guardian/models/Alert.kt
package com.aws.guardian.models

import com.google.firebase.database.IgnoreExtraProperties
import java.io.Serializable

@IgnoreExtraProperties
data class Alert(
    val id: String = "",
    val severity: String = "",  // CRITICAL, HIGH, MEDIUM, LOW
    val title: String = "",
    val message: String = "",
    val timestamp: Long = 0L,
    val account: String = "",
    val isRead: Boolean = false
) : Serializable {
    enum class Severity(val value: String) {
        CRITICAL("CRITICAL"),
        HIGH("HIGH"),
        MEDIUM("MEDIUM"),
        LOW("LOW")
    }
}

data class AlertListResponse(
    val alerts: List<Alert> = emptyList(),
    val total: Int = 0
)
